### PR TITLE
Add Get[ChildResources] methods to each parent [Resource]Operations

### PR DESCRIPTION
**Pull request by HarveyLink**

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs [hunk 1]
```diff
@@ -83,6 +87,23 @@ private void WriteClientMethods(CodeWriter writer, ResourceOperation resourceOpe
                 writer.Line($"return await ListAvailableLocationsAsync(ResourceType, cancellationToken);");
             }
             writer.Line();
+
+            foreach (var item in context.Library.ChildParent)
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs:91]: this should loop through context.Library.OperationGroups

If operationGroup.ParentResourceType == resourceOperation.OperationGroup.ResourceType

write the method.  You shouldn't need the parentchild thing you did

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs [hunk 2]
```diff
@@ -54,6 +54,8 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
 
         public IEnumerable<ResourceContainer> ResourceContainers => EnsureResourceContainers().Values;
 
+        public IDictionary<KeyValuePair<OperationGroup, ResourceOperation>, KeyValuePair<OperationGroup, ResourceOperation>> ChildParent => GetParent();
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:57]: should not be needed see comments below

**Review by m-nash — CHANGES_REQUESTED** [review]

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs (hunk 2 above)
Comment (HarveyLink) [reply src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs]: Done

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs (hunk 1 above)
Comment (HarveyLink) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs]: Done

File: samples/Azure.ResourceManager.Sample/Generated/DedicatedHostGroupOperations.cs [hunk 3]
```diff
@@ -59,5 +59,12 @@ public async Task<IEnumerable<LocationData>> ListAvailableLocationsAsync(Cancell
         {
             return await ListAvailableLocationsAsync(ResourceType, cancellationToken);
         }
+
+        /// <summary> Gets a list of DedicatedHost in the DedicatedHostGroup. </summary>
+        /// <returns> An object representing collection of DedicatedHosts and their operations over a DedicatedHostGroup. </returns>
+        public DedicatedHostContainer GetDedicatedHost()
```
Comment (m-nash) [inline samples/Azure.ResourceManager.Sample/Generated/DedicatedHostGroupOperations.cs:65]: should be plural

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs [hunk 4]
```diff
@@ -94,6 +97,21 @@ private void WriteClientMethods(CodeWriter writer, ResourceOperation resourceOpe
...
             writer.Line();
+
+            foreach (var item in context.CodeModel.OperationGroups)
+            {
+                if (item.ParentResourceType(context.Configuration.MgmtConfiguration).Equals(resourceOperation.OperationGroup.ResourceType(context.Configuration.MgmtConfiguration)))
+                {
+                    var container = context.Library.ResourceContainers.First(x => x.ResourceName.Equals(item.Resource(context.Configuration.MgmtConfiguration)));
+                    writer.WriteXmlDocumentationSummary($"Gets a list of {container.ResourceName} in the {resourceOperation.ResourceName}.");
+                    writer.WriteXmlDocumentationReturns($"An object representing collection of {container.ResourceName}s and their operations over a {resourceOperation.ResourceName}.");
+                    using (writer.Scope($"public {container.Type} Get{container.ResourceName}()"))
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs:108]: need to pluralize the name

GetPolicy => GetPolicies
GetHost => GetHosts
etc

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs [hunk 5]
```diff
@@ -291,6 +291,28 @@ public void ValidateSubscriptionExtensionsListResourceByName()
             }
         }
 
+        [Test]
+        public void ValidateParentResourceOperation()
+        {
+            foreach (var operation in FindAllOperations())
+            {
+                var operationObj = Activator.CreateInstance(operation, true);
```
Comment (m-nash) [inline test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs:299]: I don't think you need to construct one, you can simply inspect the Type itself right?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: samples/Azure.ResourceManager.Sample/Generated/DedicatedHostGroupOperations.cs (hunk 3 above)
Comment (HarveyLink) [reply samples/Azure.ResourceManager.Sample/Generated/DedicatedHostGroupOperations.cs]: Done

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs (hunk 4 above)
Comment (HarveyLink) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs]: Done

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs (hunk 5 above)
Comment (HarveyLink) [reply test/AutoRest.TestServer.Tests/Mgmt/TestProjects/TestProjectTests.cs]: Done

**Review by chamons — COMMENTED** [review]
No shared code changes, I'll leave the mgmt diffs to folks on your team.

I don't see anything concerning beyond Pluralization always being a pain to get "good enough".

**Review by m-nash — APPROVED** [review]

**Pull request merged**